thewalrus/tests/test_fock_gradients.py

```diff
@@ -148,25 +125,122 @@ def test_Kgate():
     assert np.allclose(dR, dRfd, atol=5e-4, rtol=0)
 
 
-def test_S2gate_real():
-    """Tests the value of the analytic gradient for the S2gate_real against finite differences"""
-    cutoff = 10
-    s = np.arcsinh(1.0)
-    _, dS2 = S2gate_real(s, cutoff, grad=True)
-    ds = 0.0001
-    S2s, _ = S2gate_real(s + ds, cutoff)
-    S2m, _ = S2gate_real(s - ds, cutoff)
-    dS2fd = (S2s - S2m) / (2 * ds)
-    assert np.allclose(dS2, dS2fd, atol=1e-5, rtol=0)
 
 
-def test_BSgate_real():
-    """Tests the value of the analytic gradient for the BSgate_real against finite differences"""
-    theta = 1.0
-    cutoff = 9
-    _, dBS = BSgate_real(theta, cutoff, grad=True)
-    dtheta = 0.0001
-    BSs, _ = BSgate_real(theta + dtheta, cutoff)
-    BSm, _ = BSgate_real(theta - dtheta, cutoff)
-    dBSfd = (BSs - BSm) / (2 * dtheta)
-    assert np.allclose(dBS, dBSfd, atol=1e-5, rtol=0)
+
+def test_Dgate_values(tol):
+    """Tests the correct construction of the single mode displacement operation"""
+    cutoff = 5
+    alpha = 0.3 + 0.5 * 1j
+    # This data is obtained by using qutip
+    # np.array(displace(40,alpha).data.todense())[0:5,0:5]
+    expected = np.array(
+        [
+            [0.84366482 + 0.00000000e00j, -0.25309944 + 4.21832408e-01j, -0.09544978 - 1.78968334e-01j, 0.06819609 + 3.44424719e-03j, -0.01109048 + 1.65323865e-02j,],
+            [0.25309944 + 4.21832408e-01j, 0.55681878 + 0.00000000e00j, -0.29708743 + 4.95145724e-01j, -0.14658716 - 2.74850926e-01j, 0.12479885 + 6.30297236e-03j,],
+            [-0.09544978 + 1.78968334e-01j, 0.29708743 + 4.95145724e-01j, 0.31873657 + 0.00000000e00j, -0.29777767 + 4.96296112e-01j, -0.18306015 - 3.43237787e-01j,],
+            [-0.06819609 + 3.44424719e-03j, -0.14658716 + 2.74850926e-01j, 0.29777767 + 4.96296112e-01j, 0.12389162 + 1.10385981e-17j, -0.27646677 + 4.60777945e-01j,],
+            [-0.01109048 - 1.65323865e-02j, -0.12479885 + 6.30297236e-03j, -0.18306015 + 3.43237787e-01j, 0.27646677 + 4.60777945e-01j, -0.03277289 + 1.88440656e-17j,],
+        ]
+    )
```

Comment:
Wouldn't it be a good idea to include qutip in the test files? It makes it a bit clearer (and might be useful for further tests).

Reply:
I rather not. Qutip is big package, and we don't want to add more dependencies than the essential

Reply:
Understandable. :+1:

Reply:
Yep. One of the issues with QuTiP is that it contains C++ code, and they do not provide binaries for Windows/MacOS, so it is quite hard to install.